### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by ivchenkodima**
- [x] Change names of area and region
- [x] Replace this.props.history.goBack() with this.props.router.goBack() in order to avoid troubleshooting with undefined history
- [x] Find api  troubleshooting with id of pending_declaration_request. Add tested after api had fixed that.
- [x] Add padding in DataList
- [x] Add checking for empty list in goBack() flow
- [x] Safari tested
- [x] Add cursor: pointer for BackLick icon component
- [x] Add checking for empty employee.doctor in EmployeeDetailPage
- [x] Add checking for empty declaration.employee.party in DeclarationDetailPage
- [x] Fix safari 7.1 bug with flex-box in SearchForm.

https://github.com/edenlabllc/ehealth.api/issues/728

### CODE REVIEW

**Comment by alexeybondarenko** [comment]
@ivchenkodima I've found several new bugs:

- Error in this page `http://localhost:8080/employees/5de2c7a2-6624-11e7-a151-0a580a00003a/`
Log in server 
```
TypeError: Cannot read property 'educations' of undefined
    at t.value (/Users/user/Sites/nhs.admin.web/static/server.js:1:112619)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:785:34)
    at ReactCompositeComponentWrapper._renderValidatedComponent (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:811:32)
    at ReactCompositeComponentWrapper.performInitialMount (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:353:30)
    at ReactCompositeComponentWrapper.mountComponent (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:241:21)
    at Object.mountComponent (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactReconciler.js:49:35)
    at ReactCompositeComponentWrapper.performInitialMount (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:367:34)
    at ReactCompositeComponentWrapper.mountComponent (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:241:21)
    at Object.mountComponent (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactReconciler.js:49:35)
    at ReactCompositeComponentWrapper.performInitialMount (/Users/user/Sites/nhs.admin.web/node_modules/react/lib/ReactCompositeComponent.js:367:34)
```
I've already debugged this issue (but it can be another case). In some cases employee don't have `doctor` property

- I've had an error while clicking on the back link in /declarations/:id page. I think the problem could be in the smaller data format of `declarations/` comparing with `declarations/:id`. (we talked about this yesterday)

- Add please `cursor: pointer` to the back arrow link. 
image

- I think we need to fix this behavior in Safari.
image

File: app/common/components/DataList/styles.scss [hunk 1]
```diff
@@ -18,6 +18,7 @@
   dt,
   dd {
     float: left;
+    padding-left: 5px;
```
Comment (alexeybondarenko) [inline app/common/components/DataList/styles.scss]: Are you sure that you want to add padding for `dt` also?

Comment (ivchenkodima) [inline app/common/components/DataList/styles.scss]: add only for dt

**Comment by ivchenkodima** [comment]
Tested on Safari 7.1

**Comment by ivchenkodima** [comment]
@alexeybondarenko done

**Pull request merged**